borg/helpers.py

```diff
@@ -156,33 +161,43 @@ def update(self, size, csize, unique):
         if unique:
             self.usize += csize
 
-    def print_(self, label, cache):
-        buf = str(self) % label
-        buf += "\n"
-        buf += str(cache)
-        return buf
-
-    def __str__(self):
-        return format(self, """\
+    summary = """\
                        Original size      Compressed size    Deduplicated size
-%-15s {0.osize:>20s} {0.csize:>20s} {0.usize:>20s}""")
+{label:15} {stats.osize_fmt:>20s} {stats.csize_fmt:>20s} {stats.usize_fmt:>20s}"""
+    def __str__(self):
+        return self.summary.format(stats=self, label='This archive:')
+
+    def __repr__(self):
+        fmt = "<{cls} object at {hash:#x} ({self.osize}, {self.csize}, {self.usize})>"
+        return fmt.format(cls=type(self).__name__,
+                          hash=id(self),
+                          self=self)
+
+    @property
+    def osize_fmt(self):
+        return format_file_size(self.osize)
+
+    @property
+    def usize_fmt(self):
+        return format_file_size(self.usize)
 
-    def __format__(self, format_spec):
-        fields = ['osize', 'csize', 'usize']
-        FormattedStats = namedtuple('FormattedStats', fields)
-        return format_spec.format(FormattedStats(*map(format_file_size, [ getattr(self, x) for x in fields ])))
+    @property
+    def csize_fmt(self):
+        return format_file_size(self.csize)
 
-    def show_progress(self, item=None, final=False):
+    def show_progress(self, item=None, final=False, stream=None):
+        (columns, lines) = get_terminal_size((80, 24))
```

Comment:
actually... i did that, but now i realise the fallback won't apply in 3.4/3.3.. .so maybe we need to pass it anyways down there?

i was hoping to avoid that duplication... damn you 3.2... :)